app/controllers/admin/admin_users_controller.rb

```diff
@@ -39,76 +39,115 @@ def update_user
+          @user.fannish_next_of_kin.kin_email = params[:next_of_kin_email]
+          success_message << ts('Fannish next of kin email updated.')
+        end
+        
+        # delete the next of kin if the fields are blank and changed
+        if params[:next_of_kin_user].blank? && params[:next_of_kin_email].blank? && @user.fannish_next_of_kin
+          @user.fannish_next_of_kin.destroy
+          success_message << ts('Fannish next of kin removed.')
+        end
+
+        # create warning
+        if params[:admin_action] == 'warn'
+          @user.create_log_item( options = {:action => ArchiveConfig.ACTION_WARN, :note => @admin_note, :admin_id => current_admin.id})
+          success_message << ts('Warning was recorded.')
+        end
+
+        # create suspension
+        if params[:admin_action] == 'suspend'
+          @user.suspended = true
+          @suspension_days = params[:suspend_days].to_i
+          @user.suspended_until = @suspension_days.days.from_now
+          @user.create_log_item( options = {:action => ArchiveConfig.ACTION_SUSPEND, :note => @admin_note, :admin_id => current_admin.id, :enddate => @user.suspended_until})
+          success_message << ts('User has been temporarily suspended.')
+        end
+
+        # create ban
+        if params[:admin_action] == 'ban'
           @user.banned = true
-          if @user.save && @user.banned?
-            @user.create_log_item( options = {:action => ArchiveConfig.ACTION_BAN, :note => @admin_note, :admin_id => current_admin.id})
-            flash[:notice] = t('success_banned', :default => "User has been permanently suspended")
-            redirect_to(request.env["HTTP_REFERER"] || root_path)
-          else
-            flash[:error] = t('error_banned', :default => "User could not be permanently suspended")
-            redirect_to(request.env["HTTP_REFERER"] || root_path)
-          end
-        elsif params[:admin_action] == 'unsuspend'
-          if @user.suspended?
-            @user.suspended = false
-            @user.suspended_until = nil
-            if @user.save && !@user.suspended? && @user.suspended_until.blank?
-              @user.create_log_item( options = {:action => ArchiveConfig.ACTION_UNSUSPEND, :note => @admin_note, :admin_id => current_admin.id})
-              flash[:notice] = t('success_unsuspend', :default => "Suspension has been lifted")
-              redirect_to(request.env["HTTP_REFERER"] || root_path)
-            else
-              flash[:error] = t('error_unsuspend', :default => "Suspension could not be lifted")
-              redirect_to(request.env["HTTP_REFERER"] || root_path)
-            end
-          else
-            flash[:notice] = t('not_suspended', :default => "User had not been suspended")
-            redirect_to(request.env["HTTP_REFERER"] || root_path)
+          @user.create_log_item( options = {:action => ArchiveConfig.ACTION_BAN, :note => @admin_note, :admin_id => current_admin.id})
```

Comment:
Useless assignment to variable - `options`.<br>Line is too long. [134/80]<br>Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside } missing.<br>Space inside parentheses detected.